### ISSUE TITLE
Islandora Entities CSV Import - imports identifier as type = "u1", should be type = "local" or type = "orcid"

### ISSUE THREAD

**Issue by wrandtkeflvc** [FLVC/flvc#56]
Islandora Entities has functionality to batch import a group of scholar Person Entities from a CSV file.

It converts the following columns into MADS fields
    STATUS
    POSITION
    EMAIL
    IDENTIFIER
    TERM_OF_ADDRESS
    GIVEN_NAME
    FAMILY_NAME
    FAX
    PHONE
    DISPLAY_NAME
    DEPARTMENT
    NAME_DATE
    STREET
    CITY
    STATE
    COUNTRY
    POSTCODE
    START_DATE
    END_DATE
    ROOM_NUMBER
    BUILDING
    CAMPUS

The IDENTIFIER column is converted to MADS identifier type="u1".  So far, FLVC has recommended using email address as a default identifier scheme, and using ORCID when the person has an ORCID.

To facilitate use of a local scheme which can be replaced with ORCID as various departments and faculty register for that system, the batch import of Person Entities should have the IDENTIFIER field convert to a MADS field: identifier type="local".

This is part of FLIIRT Phase 2.
and an ORCID_IDENTIFIER field convert to a MADS field: identifier type="orcid"

**Comment by wrandtkeflvc** [comment]
(I've used pre tags to try and make all the XML show up in this.  If it looks like a chunk of text is missing, then it's because git hub issues won't display info in brackets and so the XML is throwing it off.  Gail, I'll send this to you as a text file as well.)

Here's the code change to make to Islandora Entities CSV Import, and other code in Islandora which uses identifiers to link objects together:

Right now, batch csv import uses the following columns:
STATUS
POSITION
EMAIL
IDENTIFIER
TERM_OF_ADDRESS
GIVEN_NAME
FAMILY_NAME
FAX
PHONE
DISPLAY_NAME
DEPARTMENT
NAME_DATE
STREET
CITY
STATE
COUNTRY
POSTCODE
START_DATE
END_DATE
ROOM_NUMBER
BUILDING
CAMPUS

Right now, IDENTIFIER maps to MADS <pre><identifier type="u1"></pre>.  It's fine for that mapping to stay; FLVC just won't include IDENTIFIER in the Excel template provided to institutions.

Gail, could you:

1)

Add a new mapping: LOCAL_IDENTIFIER to map to MADS <pre><identifier type="local"></pre>

2)

Add a new mapping: ORCID_IDENTIFIER to map to MADS <pre><identifier type="orcid"></pre>

3)

Change the theme display of person entities, so that MADS <pre><identifier type="orcid"></pre> shows up in the display, but identifiers of other types do not show up.

4)

Check that linking of MADS <pre><identifier></pre> does not depend on type.
In other words, if a content object has a MADS field <pre><name type="personal"><nameIdentifier></pre> of any type, then that should link to any Person with a matching MADS <pre><identifier></pre> of any type.

------------------------------

Meanwhile, Mina will change the MADS form to split out identifier into type "local" and type "orcid", so that librarians can edit the info once it's loaded.

**Comment by glewisorg** [comment]
Mapping for LOCAL_IDENTIFIER and ORCID_IDENTIFIER have been added on dev and test.

I don't understand what you mean for #3 when you say "MADS shows up in the display but identifiers of other types do not".

**Comment by glewisorg** [comment]
Oh wait, this is something new, not just a csv issue.  You want to have new types of identifers in the MADS for person objects?
This would require some work - changes to gsearch indexing, code, and views.

**Comment by wrandtkeflvc** [comment]
I think that this is resolved.  See CRM ticket no. CAS-85019-S8T8 https://flvc.crm.dynamics.com:443/main.aspx?etc=112&id=0c66683c-4c62-e711-8101-5065f38a2971&histKey=206082640&newWindow=true&pagetype=entityrecord .  I'll try and confirm at the next developers meeting that it's resolved, and then close this github issue.

**Comment by wrandtkeflvc** [comment]
This issue is resolved and code is on production.  This was confirmed at the Sept. 7, 2017 Islandora Developers Meeting.
